app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -125,60 +153,63 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         )
     }
 
+    private fun onSyncNow() {
+        lifecycleScope.launch {
+            requireComponents.analytics.metrics.track(Event.SyncAccountSyncNow)
+            // Trigger a sync.
+            requireComponents.backgroundServices.accountManager.syncNowAsync().await()
+            // Poll for device events.
+            accountManager.authenticatedAccount()
+                ?.deviceConstellation()
+                ?.refreshDeviceStateAsync()
+                ?.await()
+        }
+    }
+
+    private fun makeSnackbar(newValue: String): Boolean {
+        // The network request requires a nonempty string, so don't persist any changes if the user inputs one.
+        if (newValue.trim().isEmpty()) {
+            FenixSnackbar.make(view!!, FenixSnackbar.LENGTH_LONG)
+                .setText(getString(R.string.empty_device_name_error))
+                .show()
+            return false
+        }
+        return true
+    }
+
+    private fun syncDeviceName(newValue: String) {
+        // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
+        lifecycleScope.launch(IO) {
```

Comment:
This could be launched on `Main`, since the `accountManager` will already perform actual work on the `IO` thread pool.

Reply:
Alternatively, unless you actually look at the result of `setDeviceNameAsync` (success/failure boolean flag), you don't need coroutines here at all. Just ignore the async result.